pkg/daemon/ceph/client/pool.go

```diff
@@ -406,6 +412,12 @@ func CreateReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterI
 		return errors.Wrapf(err, "failed to create replicated pool %s. %s", poolName, string(output))
 	}
 
+	args = []string{"pool", "init", poolName}
+	output, err = NewRBDCommand(context, clusterInfo, args).Run()
+	if err != nil {
+		return errors.Wrapf(err, "failed to initialize replicated pool %q. %s", poolName, string(output))
+	}
```

Comment:
Is `CreateReplicatedPoolForApp` called only for block (i.e. rbd) pools?  If not, I think this should be conditioned on `appname` being `rbd` or something like that.  And then you could move `setCommonPoolProperties(..., appname)` call into an else branch since `rbd pool init` would set the application.